### DISCUSSION TITLE
NTP overheads with RTOS

### DISCUSSION THREAD

**Discussion by RadianM** [espressif/arduino-esp32#7082] [Q&A]
I've been trying to trace why a program with several running tasks has one low priority task that never gets serviced. I found by a process of elimination that if I simply comment out the line:   **configTime(gmtOffset_sec, daylightOffset_sec, ntpServer)** everything works as I would expect, except for the time being wrong (obviously). All my tasks yield at some point and let execution take place at all priority levels so long as I don't call configTime(). I even tried calling it from a lowest priority task of its own but that made no difference.

I haven't had any success trying to find the inner workings of NTP for arduino-esp32 as some important links seem to be to esp8266 code and some are broken. Is anyone familiar with the way NTP time is handled and how it might interfere with the task scheduler?

**Comment by markxoe** [comment]
The function you are looking for is defined here: https://github.com/espressif/arduino-esp32/blob/c93bf11f1e3c3ca8088b837525dcdd78c8579a7a/cores/esp32/esp32-hal-time.c#L48-L61
And I suspect the `esp_netif_init();` to crash your program as 
> this function should be called exactly once from application code, when the application starts up

(See IDF docs), but the function is already called for the WiFi stuff:
https://github.com/espressif/arduino-esp32/blob/c93bf11f1e3c3ca8088b837525dcdd78c8579a7a/libraries/WiFi/src/WiFiGeneric.cpp#L618

You may need to create an issue and/or PR for this...
